### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by AnnaaKorzhenko**
implemented few sproc (IN, OUT and INOUT) and used transaction with a commit logic

### CODE REVIEW

File: pa4procedures.sql [hunk 1]
```diff
@@ -0,0 +1,75 @@
+----------- IN -----------------------
+-- it takes a team name as an IN parameter and selects all the information about that exact team
+-- creating a procedure
+DELIMITER &&
+
+CREATE PROCEDURE GetTeamInfo (IN teamName VARCHAR(255))
+BEGIN
+    SELECT * FROM teams WHERE team_name = teamName;
```
Comment (Ivashchyk) [inline pa4procedures.sql]: Do not use * in production code

File: pa4procedures.sql [hunk 2]
```diff
@@ -0,0 +1,75 @@
+----------- IN -----------------------
+-- it takes a team name as an IN parameter and selects all the information about that exact team
+-- creating a procedure
+DELIMITER &&
+
+CREATE PROCEDURE GetTeamInfo (IN teamName VARCHAR(255))
```
Comment (Ivashchyk) [inline pa4procedures.sql]: Rename to get_team_info_sproc

File: pa4procedures.sql [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+END&&
+ 
+DELIMITER ;
+
+
+----------- OUT ------------------------
+-- it gives total count of existing players as an output
+DELIMITER &&
+
+CREATE PROCEDURE GetPlayerCount (OUT totalPlayers INT)
```
Comment (Ivashchyk) [inline pa4procedures.sql]: Rename to get_player_count_sproc

File: pa4procedures.sql [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+BEGIN
+SELECT COUNT(*) INTO totalPlayers FROM players;
```
Comment (Ivashchyk) [inline pa4procedures.sql]: COUNT(1)

File: pa4procedures.sql [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+END&&
+DELIMITER ;
+   
+SET @totalPLayers = NULL;
```
Comment (Ivashchyk) [inline pa4procedures.sql]: I guess it is a duplicate from the executions file. If yes then remove it

File: pa4procedures.sql [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+CALL GetPlayerCount(@totalPlayers);
+SELECT @totalPlayers;
+   
+
+
+---------- INOUT ------------------------
+-- it takes player id as an IN parameter and then returns number of goals scored bt this player as an OUT parameter, but using INOUT
+DELIMITER &&
+
+CREATE PROCEDURE CalculatePlayerGoals (
```
Comment (Ivashchyk) [inline pa4procedures.sql]: rename to calculate_player_goals_sproc

File: pa4procedures.sql [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    INOUT playerID INT
+)
+BEGIN
+    DECLARE totalGoals INT;
```
Comment (Ivashchyk) [inline pa4procedures.sql]: rename to total_goals

File: pa4procedures.sql [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+SELECT COUNT(*) INTO totalGoals FROM goals WHERE player_id = playerID;
```
Comment (Ivashchyk) [inline pa4procedures.sql]: COUNT(1)

File: pa4procedures.sql [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+   
+SET @param = 5;
+CALL CalculatePlayerGoals(@param);
+SELECT @param;
+   
+
+------------- TRANSACTION -------------------------
+-- if the input data about match's goals is accurate (not negative), it updates the info in "matches" table and commits
+-- else, it still updates, but after the "if" condition, it does rollback
+CREATE PROCEDURE update_match_result(
```
Comment (Ivashchyk) [inline pa4procedures.sql]: rename to update_match_result_sproc

**Review by Ivashchyk — CHANGES_REQUESTED** [review]
minor issues

**Comment by AnnaaKorzhenko** [comment]
Hello! I have updated my assignment due to your recommendations and
requested changes. I changed the naming and avoided using *
Here's the needed commit in the PR:
https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3/commits/6a098597988ba1a600e7f869de3ff500498e6bcb

**Pull request merged**